extension.js

```diff
@@ -1,59 +1,47 @@
-// The module 'vscode' contains the VS Code extensibility API
-// Import the module and reference it with the alias vscode in your code below
 const vscode = require('vscode');
 const cp = require('child_process')
 
 let runCommand = function (command) {
 	cp.exec(command, (err, stdout, stderr) => {
 		if (err) {
-			console.log('error: ' + err);
+			logError(err);
 		}
 	});
 }
 
-// let figlog = function (vscode, console) {
-// 	console.log(vscode.window.activeTerminal, vscode.window.activeTextEditor, vscode.context)
-// 	console.log(vscode.window.activeTerminal.name)
-// 		(async () => {
-// 			let pid = await vscode.window.activeTerminal.processId
-// 			console.log(pid)
-// 		})()
-// }
+// fig.log is an internal setting as it's intended for developers only, this means it won't show up
+// in the settings UI/editor. Add `"fig.log": true` to your settings.json and reload the window to
+// enable logging.
+let shouldLog = vscode.workspace.getConfiguration().get('fig.log') === true;
 
-// this method is called when your extension is activated
-// your extension is activated the very first time the command is executed
+function log(...args) {
+	if (shouldLog) {
+		console.log(`fig: ${args[0]}`, args.slice(1));
+	}
+}
 
-/**
- * @param {vscode.ExtensionContext} context
- */
+function logError(message) {
+	console.error(`fig: ${message}`);
+}
 
 function activate(context) {
-	// var next = vscode.window.terminals.length
-	// var uniqueTerminalIDs = vscode.window.terminals.reduce((elm, acc, idx) => {
-	// 	acc[elm] = idx
-	// 	return acc
-	// })
 	try {
-		console.log("Terminals: " + vscode.window.terminals.length);
+		log("Terminals: " + vscode.window.terminals.length);
 
 		try {
-			console.log(vscode.window.activeTerminal.name)
+			log(vscode.window.activeTerminal.name)
 			vscode.window.activeTerminal.processId.then((processId) => {
 				if (processId) {
 					runCommand('fig bg:vscode code:' + processId)
 				}
 			})
 		} catch (e) {
-			console.log("Fig integration: " + e)
+			log("Fig integration: " + e)
 		}
 
-		// vscode.window.onDidOpenTerminal
 		vscode.window.onDidChangeWindowState(event => {
-			console.log("onDidChangeWindowState", event)
-			console.log(vscode.window.activeTerminal, vscode.window.activeTextEditor, vscode.context)
-			if (!state.focused) {
-				return
-			}
```

Comment:
This part might need testing, I believe this would always throw before so the below code will now run.